src/main/scala/io/shiftleft/fuzzyc2cpg/passes/cfgcreation/CfgCreator.scala

```diff
@@ -0,0 +1,392 @@
+      case "DoStatement" =>
+        cfgForDoStatement(node)
+      case "ForStatement" =>
+        cfgForForStatement(node)
+      case "GotoStatement" =>
+        cfgForGotoStatement(node)
+      case "IfStatement" =>
+        cfgForIfStatement(node)
+      case "ElseStatement" =>
+        cfgForChildren(node)
+      case "SwitchStatement" =>
+        cfgForSwitchStatement(node)
+      case _ =>
+        Cfg.empty
+    }
+
+  /**
+    * The CFG for a break/continue statements contains only
+    * the break/continue statement as a single entry node.
+    * The fringe is empty, that is, appending
+    * another CFG to the break statement will
+    * not result in the creation of an edge from
+    * the break statement to the entry point
+    * of the other CFG.
+    * */
+  private def cfgForBreakStatement(node: nodes.ControlStructure): Cfg =
+    Cfg(Some(node), breaks = List(node))
+
+  private def cfgForContinueStatement(node: nodes.ControlStructure): Cfg =
+    Cfg(Some(node), continues = List(node))
+
+  /**
+    * Jump targets ("labels") are included in the CFG. As these
+    * should be connected to the next appended CFG, we specify
+    * that the label node is both the entry node and the only
+    * node in the fringe. This is achieved by calling `cfgForSingleNode`
+    * on the label node. Just like for breaks and continues, we record
+    * labels. We store case/default labels separately from other labels,
+    * but that is not a relevant implementation detail.
+    * */
+  private def cfgForJumpTarget(n: nodes.JumpTarget): Cfg = {
+    val labelName = n.name
+    val cfg = cfgForSingleNode(n)
+    if (labelName.startsWith("case") || labelName.startsWith("default")) {
+      cfg.copy(caseLabels = List(n))
+    } else {
+      cfg.copy(labeledNodes = Map(labelName -> n))
+    }
+  }
+
+  /**
+    * A CFG for a goto statement is one containing the goto
+    * node as an entry node and an empty fringe. Moreover, we
+    * store the goto for dispatching with `withResolvedGotos`
+    * once the CFG for the entire method has been calculated.
+    * */
+  private def cfgForGotoStatement(node: nodes.ControlStructure): Cfg = {
+    // TODO: the goto node should contain a field for the target so that
+    // we can avoid the brittle split/slice operation here
+    val target = node.code.split(" ").lastOption.map(x => x.slice(0, x.length - 1))
+    target.map(t => Cfg(Some(node), gotos = List((node, t)))).getOrElse(Cfg.empty)
+  }
+
+  /**
+    * Return statements may contain expressions as return values,
+    * and therefore, the CFG for a return statement consists of
+    * the CFG for calculation of that expression, appended to
+    * a CFG containing only the return node, connected with
+    * a single edge to the method exit node. The fringe is
+    * empty.
+    * */
+  private def cfgForReturn(actualRet: nodes.Return): Cfg = {
+    cfgForChildren(actualRet) ++
+      Cfg(Some(actualRet), singleEdge(actualRet, exitNode), List())
```

Comment:
Since we care a lot about readability for this code, I suggest we always use named arguments when creating a `Cfg` object. They would affect a lot of places but would help for a better understanding.